pynecone/state.py

```diff
@@ -45,6 +45,9 @@ class State(Base, ABC):
     # The set of dirty substates.
     dirty_substates: Set[str] = set()
 
+    # The routing path that triggered the state
+    router_data: Dict[str, Any]
```

Comment:
We can default this to an empty dictionary:
```
router_data: Dict[str, Any] = {}
```

Also, we need to update:
* this file on line 90, add `router_data` to the set of `skip_vars`
* `tests/test_state.py`, on line 151, add `router_data` to the tuple of fields to get the unit test passing